### PR TITLE
Reddit parsing

### PR DESCRIPTION
**Pull request by Zasheus**
## Todo

- [x] implement a class for parsing reddit data

## Attached GitHub issue

Closes https://github.com/se310-t6/politicry/issues/16

### Local Build
- [x] Ran all pre-commit hooks `pre-commit run --all-files`
- [x] Extension has been tested to build correctly `cd extension && yarn && yarn build`
- [x] Backend has been tested to build correctly `cd backend && docker-compose build`
- [x] Extension test suite has been run locally `cd extension && yarn && yarn test`
- [x] Backend test suite has been run locally `cd backend && python3 -m unittest`

### GitHub
- [x] Target branch has been set correctly
- [x] PR has been rebased onto target branch
- [x] PR has been assigned an owner
- [x] Author has performed a self-review of the code
- [x] Removed the WIP message from the top of this PR

### CODE REVIEW

**Comment by atan958** [comment]
As previously suggested, I'd recommend performing the pre-commit stuff on your local repo https://pre-commit.com/

**Review by atan958 — APPROVED** [review]
It looks to be good as far as I can tell. Great work Ryan!
All I can say is that some changes may need to be done to the original extension UI given we have only working functionality for Reddit, but that is work for another issue.

File: extension/public/manifest.json [hunk 1]
```diff
@@ -12,5 +12,12 @@
         "activeTab",
         "scripting"
     ],
-    "version": "1.0"
+    "version": "1.0",
+    "content_scripts": [
+        {
+          "matches": ["https://www.reddit.com/*", "https://www.imgur.com/*"],
+          "js": ["jquery-1.10.2.min.js", "reddit.js"],
```
Comment (JosiahBull) [inline extension/public/manifest.json:19]: Do we need to use jquery? Pretty chonky library if we can get away without it.

**Review by lche262 — APPROVED** [review]
lgtm!

**Review by hmin697 — APPROVED** [review]
great!!!

**Pull request merged**